### PR TITLE
debt/MET-4723 Pattern decade date extractor cleanup

### PR DESCRIPTION
**Pull request by stzanakis**

### CODE REVIEW

File: metis-normalization/src/test/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractorTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        of("180u", "180X", DECADE),
+        of("180X", "180X", DECADE),
+        of("180U", "180X", DECADE),
+        of("  180u  ", "180X", DECADE),
+        of("180x?", "180X?", DECADE),
+        of("180u?", "180X?", DECADE),
+        of("180??", "180X?", DECADE),
+        of("?180x", "180X?", DECADE),
+        of("?180u", "180X?", DECADE),
+        of("?180u?", "180X?", DECADE),
```
Comment (jeortizquan) [inline metis-normalization/src/test/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractorTest.java:51]: Derived from the regex new test cases came up.
```
        of("?180x?", "180X?", DECADE),
        of("?180??", "180X?", DECADE),

        //just to validate that for 3 digits the decade is extracted at this moment.
        of("?1280x", null, null),
        of("?1280u?", null, null),
        of("?1280??", null, null),
        of("1280??", null, null),
```

**Review by jeortizquan — COMMENTED** [review]
I think we are good, some test case to consider if they apply, and some questions.

File: metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractor.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ *     <li>180x?</li>
+ *   </ul>
+ * </p>
+ * <p>
+ * A decade represented as YYYu or YYYx. For example, '198u', '198x' Dates such as '198-' are not supported because they may
+ * indicate a decade or a time period with an open end
+ */
+public class DecadeDateExtractor implements DateExtractor {
+
+  static final Pattern decadePattern = Pattern.compile("\\??(\\d{3})(?:[ux]\\??|\\?\\?)", Pattern.CASE_INSENSITIVE);
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractor.java:33]: No it must be private.

File: metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractor.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Extractor that matches a decade of the format YYY[ux].
+ * <p>The decade may start and/or end with a question mark indicating uncertainty. Dates such as '198-' are not supported because
+ * they may indicate a decade or a time period with an open end.</p>
+ * <p>
+ * Examples:
+ *   <ul>
+ *     <li>180u</li>
+ *     <li>180x</li>
+ *     <li>180-</li>
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-normalization/src/main/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractor.java:20]: No it should just not be capture. I removed the example from the javadoc, it shouldn't be there.

File: metis-normalization/src/test/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractorTest.java (hunk 1 above)
Comment (stzanakis) [reply metis-normalization/src/test/java/eu/europeana/normalization/dates/extraction/dateextractors/DecadeDateExtractorTest.java:51]: I've added the cases you'd like

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**